temoa/extensions/stochastics/scenario_creator.py

```
@@ -0,0 +1,109 @@
+
+            # Check if filter matches
+            match = True
+            for filter_key, filter_val in p.filter.items():
+                if index_map.get(filter_key) != filter_val:
+                    match = False
+                    break
+
+            if match:
+                allowed_actions = {'multiply', 'add', 'set'}
+                if p.action not in allowed_actions:
+                    raise ValueError(f"Invalid perturbation action '{p.action}' for table {p.table}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)